apps/financial-aid/web-osk/src/components/ApplicationProvider/ApplicationProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { createContext, ReactNode, useEffect, useState } from 'react'
+
+import { Application } from '@island.is/financial-aid/shared/lib'
+
+import { GetApplicationQuery } from '@island.is/financial-aid-web/osk/graphql'
+import { useQuery } from '@apollo/client'
+import { useRouter } from 'next/router'
+import { ApolloError } from 'apollo-client'
+
+interface ApplicationProvider {
+  myApplication?: Application
+  loading: boolean
+  fetchError?: ApolloError
+}
+
+interface Props {
+  children: ReactNode
+}
+
+interface ApplicantData {
+  application?: Application
+}
+
+export const ApplicationContext = createContext<ApplicationProvider>({
+  loading: true,
+})
+
+const ApplicationProvider = ({ children }: Props) => {
+  const router = useRouter()
+
+  const storageKey = 'myCurrentApplication'
```

Comment:
I would keep this const elsewhere, either in const or above the provider definition